tests/test_nearest_neighbors.py

```diff
@@ -0,0 +1,144 @@
+from typing import Tuple
+
+import numpy as np
+import pytest
+from sklearn.datasets import make_blobs
+
+from spark_rapids_ml.knn import NearestNeighbors
+
+from .sparksession import CleanSparkSession
+from .utils import array_equal, create_pyspark_dataframe, idfn
+
+
+def test_example(gpu_number: int) -> None:
+    data = [
+        ([1.0, 1.0],),
+        ([2.0, 2.0],),
+        ([3.0, 3.0],),
+        ([4.0, 4.0],),
+        ([5.0, 5.0],),
+        ([6.0, 6.0],),
+        ([7.0, 7.0],),
+        ([8.0, 8.0],),
+    ]
+
+    query = [
+        ([0.0, 0.0],),
+        ([1.0, 1.0],),
+        ([4.1, 4.1],),
+        ([8.0, 8.0],),
+        ([9.0, 9.0],),
+    ]
+
+    topk = 2
+
+    conf = {"spark.sql.execution.arrow.maxRecordsPerBatch": str(2)}
+    with CleanSparkSession(conf) as spark:
+        schema = f"features array<float>"
+        data_df = spark.createDataFrame(data, schema)
+        query_df = spark.createDataFrame(query, schema)
+
+        gpu_knn = NearestNeighbors(num_workers=gpu_number)
+        gpu_knn = gpu_knn.setInputCol("features")
+        gpu_knn = gpu_knn.setK(topk)
+
+        gpu_knn = gpu_knn.fit(data_df)
+        distances_df, indices_df = gpu_knn.kneighbors(query_df)
+
+        import math
+
+        distances = distances_df.collect()
+        distances = [r[0] for r in distances]
+        indices = indices_df.collect()
+        indices = [r[0] for r in indices]
+
+        assert array_equal(distances[0], [math.sqrt(2.0), math.sqrt(8.0)])
+        assert indices[0] == [0, 1]
+
+        assert array_equal(distances[1], [0.0, math.sqrt(2.0)])
+        assert indices[1] == [0, 1]
+
+        assert array_equal(
+            distances[2], [math.sqrt(0.01 + 0.01), math.sqrt(0.81 + 0.81)]
+        )
+        assert indices[2] == [3, 4]
+
+        assert array_equal(distances[3], [0.0, math.sqrt(2.0)])
+        assert indices[3] == [7, 6]
+
+        assert array_equal(distances[4], [math.sqrt(2.0), math.sqrt(8.0)])
+        assert indices[4] == [7, 6]
+
+
+@pytest.mark.parametrize("feature_type", ["array"])
```

Comment:
Does this mean only 'array' input type is currently supported?

Reply:
Yeah, created a ticket to support multi-col df and vector df.